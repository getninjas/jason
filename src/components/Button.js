import React, { Component } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  children: PropTypes.string.isRequired,
  handleButtonClick: PropTypes.func.isRequired,
  isSubmit: PropTypes.bool,
};

const defaultProps = {
  children: 'Prosseguir',
  isSubmit: false,
};

export default class Button extends Component {
  render() {
    const type = this.props.isSubmit ? 'submit' : 'button';
    const style = 'btn btn--regular btn--high-contrast btn--fluid space-box-medium';

    return (
      <button type={type} className={style} onClick={this.props.handleButtonClick}>
        {this.props.children}
      </button>
    );
  }
}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;
