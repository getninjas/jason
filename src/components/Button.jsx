import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Button extends Component {
  static defaultProps = {
    children: 'Prosseguir',
    isSubmit: false,
  }

  render() {
    const type = this.props.isSubmit ? 'submit' : 'button';
    const style = "btn btn--regular btn--high-contrast btn--fluid space-box-medium";

    return (
      <button type={type} className={style} onClick={this.props.handleButtonClick}>
        {this.props.children}
      </button>
    );
  }
}

Button.propTypes = {
  children: PropTypes.string.isRequired,
  handleButtonClick: PropTypes.func.isRequired,
  isSubmit: PropTypes.bool,
}
