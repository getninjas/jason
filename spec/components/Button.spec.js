import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../src/components/Button';
import { enzymeConfig, shallow } from '../enzymeConfig';

enzymeConfig();

describe('Button', () => {
  it('renders defaultProps', () => {
    const component = renderer.create(<Button handleButtonClick={() => {}}>Test</Button>);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('when prop submit is true', () => {
    it('renders button with type=submit', () => {
      const component = shallow(<Button isSubmit={true} handleButtonClick={() => {}}>Test</Button>);

      expect(component.prop('type')).toBe('submit');
    });
  });
});
