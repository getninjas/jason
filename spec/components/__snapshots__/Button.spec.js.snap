// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders custom props 1`] = `
<button
  className="btn btn--regular btn--high-contrast btn--fluid space-box-medium"
  onClick={[Function]}
  type="button"
>
  Test
</button>
`;

exports[`Button renders defaultProps 1`] = `
<button
  className="btn btn--regular btn--high-contrast btn--fluid space-box-medium"
  onClick={[Function]}
  type="button"
>
  Test
</button>
`;
